lxc/storage_volume.go

```diff
@@ -1628,6 +1629,53 @@ func (c *cmdStorageVolumeMove) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	// Parse remote
+	resources, err := c.global.ParseServers(args[0], args[1])
+	if err != nil {
+		return err
+	}
+
+	// Source
+	srcResource := resources[0]
+	if srcResource.name == "" {
+		return fmt.Errorf(i18n.G("Missing source volume name"))
+	}
+
+	srcRemote := srcResource.remote
+	srcPath := srcResource.name
+
+	// Get source pool and volume name
+	srcVolName, srcVolPool := c.storageVolume.parseVolumeWithPool(srcPath)
+	if srcVolPool == "" {
+		return fmt.Errorf(i18n.G("No storage pool for source volume specified"))
+	}
+
+	// Destination
+	dstResource := resources[1]
+	dstRemote := dstResource.remote
+	dstPath := dstResource.name
+
+	// Get target pool and volume name
+	dstVolName, dstVolPool := c.storageVolume.parseVolumeWithPool(dstPath)
+	if dstVolPool == "" {
+		return fmt.Errorf(i18n.G("No storage pool for target volume specified"))
+	}
+
+	// Rename volume if both remotes and pools of source and target are equal.
+	if srcRemote == dstRemote && srcVolPool == dstVolPool {
+		var args []string
+
+		if srcRemote != "" {
```

Comment:
What does this bit do @monstermunchkin ?

Reply:
That's just to fill in the arguments for remote and pool correctly. `lxc storage volume rename` accepts `[<remote>:]<pool>` which is handled here.

Reply:
Do we need an equivalent for the destination too then or does it apply to both?

Reply:
> Do we need an equivalent for the destination too then or does it apply to both?

Nope. `lxc storage volume rename` only takes one remote.

Reply:
Great thanks. Please can you add a test for this new fix.